### ISSUE TITLE
Sync Batch norm stalled ranks.

### ISSUE THREAD

**Issue by jaypatravali** [horovod/horovod#2241]
**Environment:**
1. Framework:  PyTorch 
2. Framework version: 1.6.0
3. Horovod version: 0.20.0
4. MPI version:   3.1.2
5. CUDA version: 10.2    
6. NCCL version:  nccl 2.7
7. Python version: 3.7.2
8. Spark / PySpark version: 
9. OS and version: (Ubuntu 7.4.0-1ubuntu1~18.04.1)
10. GCC version:   7.4.0 
**Bug report:**
 [2020-09-05 04:24:42.562003: W /tmp/pip-install-8tt7ftw3/horovod/horovod/common/stall_inspector.cc:105] One or more tensors were submitted to be reduced, gathered or broadcasted by subset of ranks and are waiting for remainder of ranks for more than 60 seconds. This may indicate that different ranks are trying to submit different tensors or that only subset of ranks is submitting tensors, which will cause deadlock. 
Stalled ranks:
2: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
3: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
7: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
8: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
9: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
10: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
11: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
16: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
18: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
19: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
20: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
22: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
24: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
25: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]
31: [allgather.sync_batch_norm.count, allgather.sync_batch_norm.invstd, allgather.sync_batch_norm.mean]

 I am training videos on Resnet3D50 on 32 nodes (single GPUs), in my model I have replaced the 3D batch norm layers with horovod sync batch norm layers. We have v100 16GB cards which work okay without this warning/info statements on lower batch size (~20)  however we get this issue on a higher batch size (~32).

Could you please help.
